### PR TITLE
Adding token reward for adding records

### PR DESCRIPTION
**Pull request by godfreyhobbs**
When a user uploads data, 1 Finney LIN token is transferred from the Linnia admin pool/acct to the user’s address.
Currently, no data validation needed.

### CODE REVIEW

File: contracts/LinniaRecords.sol [hunk 1]
```diff
@@ -99,6 +102,33 @@ contract LinniaRecords is Ownable, Pausable, Destructible {
         return true;
     }
 
+    /// Add a record by user without any provider's signatures and get a reward.
+    ///
+    /// @param dataHash the hash of the data
+    /// @param metadata plaintext metadata for the record
+    /// @param dataUri the data uri path of the encrypted data
+    /// @param token the ERC20 token address for the rewarding token
+    function addRecordwithReward (
```
Comment (MaxBlaushild) [inline contracts/LinniaRecords.sol]: Why a separate method? Nobody will choose the original method

File: contracts/LinniaRecords.sol [hunk 2]
```diff
@@ -99,6 +102,33 @@ contract LinniaRecords is Ownable, Pausable, Destructible {
...
+        bytes32 dataHash, string metadata, string dataUri, address token)
```
Comment (MaxBlaushild) [inline contracts/LinniaRecords.sol]: Why is the token an argument? Are we expected to keep balances in many ERC20 tokens?

File: contracts/mock/ERC20Mock.sol [hunk 3]
```diff
@@ -0,0 +1,58 @@
+pragma solidity 0.4.24;
```
Comment (MaxBlaushild) [inline contracts/mock/ERC20Mock.sol]: Why not import LINToken from npm?

File: contracts/LinniaRecords.sol (hunk 1 above)
Comment (godfreyhobbs) [reply contracts/LinniaRecords.sol]: Not all use cases make sense with the reward.  For instance, a non-payable Smart contract may want to add records.

**Pull request merged**